estela-web/src/pages/ProjectDashboardPage/index.tsx

```diff
@@ -118,9 +125,87 @@ export class ProjectDashboardPage extends Component<RouteComponentProps<RoutePar
         }
 
         await this.apiService.apiStatsList(params).then(
-            (response: GlobalStats[]) => {
+            (response: ProjectStats[]) => {
+                const tempStats = new Map<string, Stats[]>();
+                response.forEach((stat) => {
+                    if (stat.stats.runtime) {
+                        stat.stats.runtime = parseDurationToSeconds(stat.stats.runtime.toString());
+                    }
+                    const tempDate = moment(stat.date).format("YYYY-MM-DD");
+                    const statsArr = tempStats.get(tempDate);
+                    if (statsArr) {
+                        tempStats.set(tempDate, statsArr.concat([{ ...stat.stats }]));
+                        return;
+                    }
+                    tempStats.set(tempDate, [{ ...stat.stats }]);
+                });
+                const finalProjectStats: ProjectStats[] = Array.from(tempStats.entries()).map(([date, stats]) => {
+                    const reducedStats = stats.reduce((acc, curr) => {
+                        return {
+                            jobs: {
+                                totalJobs: (acc.jobs?.totalJobs ?? 0) + (curr.jobs?.totalJobs ?? 0),
+                                waitingJobs: (acc.jobs?.waitingJobs ?? 0) + (curr.jobs?.waitingJobs ?? 0),
+                                runningJobs: (acc.jobs?.runningJobs ?? 0) + (curr.jobs?.runningJobs ?? 0),
+                                stoppedJobs: (acc.jobs?.stoppedJobs ?? 0) + (curr.jobs?.stoppedJobs ?? 0),
+                                completedJobs: (acc.jobs?.completedJobs ?? 0) + (curr.jobs?.completedJobs ?? 0),
+                                inQueueJobs: (acc.jobs?.inQueueJobs ?? 0) + (curr.jobs?.inQueueJobs ?? 0),
+                                errorJobs: (acc.jobs?.errorJobs ?? 0) + (curr.jobs?.errorJobs ?? 0),
+                            },
+                            pages: {
+                                totalPages: (acc.pages.totalPages ?? 0) + (curr.pages.totalPages ?? 0),
+                                scrapedPages: (acc.pages.scrapedPages ?? 0) + (curr.pages.scrapedPages ?? 0),
+                                missedPages: (acc.pages.missedPages ?? 0) + (curr.pages.missedPages ?? 0),
+                            },
+                            itemsCount: (acc.itemsCount ?? 0) + (curr.itemsCount ?? 0),
```

Comment:
Is it possible to improve this repeated actions?

Reply:
Removed everything is solved in backend